### ISSUE TITLE
tfe_registry_module resource returns "resource not found" error

### ISSUE THREAD

**Issue by fabiendelpierre** [hashicorp/terraform-provider-tfe#256]
It seems straightforward enough but it's not working for me.

I'm using a free Terraform Cloud account and my workspace uses TF v0.14.2 and version 0.23.0 of this provider.

I have an OAuth client setup for my VCS provider, which is GitHub.com. This client works fine for linking TF Cloud workspaces to Git repos and has owner privileges over the GitHub org.

My code looks like this:

```hcl
resource "tfe_registry_module" "virtual_network" {
  vcs_repo {
    display_identifier = "fabiendelpierre/terraform-azurerm-virtualnetwork"
    identifier = "fabiendelpierre/terraform-azurerm-virtualnetwork"
    oauth_token_id = data.tfe_oauth_client.my_oauth_client.oauth_token_id
  }
}
```

It's supposed to fetch this: https://github.com/fabiendelpierre/terraform-azurerm-virtualnetwork -- nothing fancy.

When the workspace runs, the plan phase says:

```
# tfe_registry_module.virtual_network will be created
  + resource "tfe_registry_module" "virtual_network" {
      + id              = (known after apply)
      + module_provider = (known after apply)
      + name            = (known after apply)
      + organization    = (known after apply)

      + vcs_repo {
          + display_identifier = "fabiendelpierre/terraform-azurerm-virtualnetwork"
          + identifier         = "fabiendelpierre/terraform-azurerm-virtualnetwork"
          + oauth_token_id     = "<oauth_token_id>" # this is the correct value, I triple checked
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

Then the apply phase returns:

```
Terraform v0.14.2
Initializing plugins and modules...
tfe_registry_module.virtual_network: Creating...

Error: Error creating registry module from repository fabiendelpierre/terraform-azurerm-virtualnetwork: resource not found

  on module_registry.tf line 1, in resource "tfe_registry_module" "virtual_network":
   1: resource "tfe_registry_module" "virtual_network" {
```

I figured I was doing something wrong, even though this matches what the [docs](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/registry_module) tell you to do, so I went through the TF Cloud GUI to add this module by hand, using the same VCS connection. Works fine.

I then pulled info about the module through the API:

```bash
$ curl -XGET \
-H "Content-Type: application/vnd.api+json" \
-H "Authorization: Bearer $TOKEN" \
https://app.terraform.io/api/v2/registry-modules/show/my-org/virtualnetwork/azurerm
```
Which returns:
```json
{
  "data": {
    "id": "mod-QhF1AMqqkNxRrfZj",
    "type": "registry-modules",
    "attributes": {
      "name": "virtualnetwork",
      "provider": "azurerm",
      "status": "setup_complete",
      "version-statuses": [
        {
          "version": "0.1.0",
          "status": "ok"
        }
      ],
      "created-at": "2020-12-16T22:25:03.317Z",
      "updated-at": "2020-12-16T22:25:03.900Z",
      "vcs-repo": {
        "branch": "",
        "ingress-submodules": true,
        "identifier": "fabiendelpierre/terraform-azurerm-virtualnetwork",
        "display-identifier": "fabiendelpierre/terraform-azurerm-virtualnetwork",
        "oauth-token-id": "<oauth_token_id>",
        "webhook-url": "https://app.terraform.io/webhooks/vcs/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
      },
      "permissions": {
        "can-delete": true,
        "can-resync": true,
        "can-retry": true
      }
    },
    "relationships": {
      "organization": {
        "data": {
          "id": "my-org",
          "type": "organizations"
        }
      }
    },
    "links": {
      "self": "/api/v2/registry-modules/show/my-org/virtualnetwork/azurerm"
    }
  }
}
```

This shows that the values I provided in my TF code for the `identifier` and `display_identifier` fields are correct, and that the OAuth token ID is also correct. So the TF code should be working just like the GUI but that's not the case.

Where did I go wrong?

**Comment by chrisarcand** [comment]
Hi @fabiendelpierre!

* How are you executing Terraform? Locally, without Terraform Cloud (but provisioning Terraform Cloud)? Using a [Terraform Cloud workspace with remote operations?](https://www.terraform.io/docs/cloud/run/index.html)
* [There are several different ways to provide an API token to this provider](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs#authentication); which are you using?

I suspect, given how thoroughly you checked your values above (thank you!), that this is likely not a problem with your repository, OAuth token, or resource block, but instead an issue with your authentication token used by the provider to interact with Terraform Cloud (you can receive `resource not found` if your request was not authenticated properly, as well as it literally not being found); so let's make sure that's working first 👍🏻

**Comment by fabiendelpierre** [comment]
https://www.terraform.io/docs/cloud/users-teams-organizations/api-tokens.html

**Comment by fabiendelpierre** [comment]
Hi @chrisarcand, thanks for your reply!

- I'm running Terraform via the TF Cloud UI/VCS-driven workflow. Indeed, using remote operations as defined in the link you sent. Basically after setting up a couple of foundational things manually in my account, I want to manage as much as possible via code.
- I'm using an org token.

It looks like ([source](https://www.terraform.io/docs/cloud/users-teams-organizations/api-tokens.html)) an org token might not be compatible with what I'm trying to do then, is that it? And I need a team or personal token?

**Comment by fabiendelpierre** [comment]
Went ahead and instantiated an additional TFE provider with an alias and a team token, and was able to import my module into the registry without code changes (other than adding a `provider` argument to my `tfe_registry_module` resource). Thanks for putting me on the right track!

**Comment by abstrask** [comment]
I experienced the same. However the token used for the `tfe` provider is being used in several other workspaces, to manage pretty much all other aspects of Terraform Cloud - but still failed on `tfe_registry_module`.

I replaced the *organization* API token with a *team* API token used for the `tfe` provider, which solved my problem.

This is subtly documented in the bottom of this table: https://developer.hashicorp.com/terraform/cloud-docs/users-teams-organizations/api-tokens#access-levels, but there's no mentioning of it on https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/registry_module.

Perhaps it would be a good idea to add a yellow "note" box at the top of the resource documentation? It could have saved me hours and lots of grey hair!
